File: lib/shared/job_posting.dart
/// Shop-owned job. Screens use this instead of a table row.
/// Marketplace cards stay on [JobListing].
class JobPosting {
  const JobPosting({
    required this.id,
    required this.title,
    required this.description,
    required this.jobType,
    required this.isActive,
    required this.createdAt,
    this.businessId,
    this.posterId,
    this.category,
    this.salaryRange,
    this.locationText,
  });

  final String id;
  final String title;
  final String description;
  final String jobType;
  final bool isActive;
  final DateTime createdAt;
  final String? businessId;
  final String? posterId;
  final String? category;
  final String? salaryRange;
  final String? locationText;

  /// Java `JobResponse`.
  factory JobPosting.fromJson(Map<String, dynamic> json) {
    final created = json['createdAt'];
    return JobPosting(
      id: '${json['id'] ?? ''}',
      title: '${json['title'] ?? ''}',
      description: '${json['description'] ?? ''}',
      jobType: '${json['jobType'] ?? ''}',
      isActive: json['active'] as bool? ?? json['isActive'] as bool? ?? true,
      createdAt: created is String
          ? DateTime.tryParse(created) ?? DateTime.fromMillisecondsSinceEpoch(0)
          : DateTime.fromMillisecondsSinceEpoch(0),
      businessId: json['businessId'] == null ? null : '${json['businessId']}',
      posterId: json['posterId'] == null ? null : '${json['posterId']}',
      category: json['category'] as String?,
      salaryRange: json['salaryRange'] as String?,
      locationText: json['locationText'] as String?,
    );
  }
}

File: degloor-one-backend/src/main/java/com/degloor/one/job/entity/JobPosting.java
package com.degloor.one.job.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "jobs")
public class JobPosting {
    @Id
    private UUID id;
    @Column(name = "business_id", nullable = false)
    private UUID businessId;
    @Column(name = "poster_id", nullable = false)
    private UUID posterId;
    @Column(nullable = false)
    private String title;
    @Column(nullable = false)
    private String description;
    private String category;
    @Column(name = "job_type", nullable = false)
    private String jobType;
    @Column(name = "salary_range")
    private String salaryRange;
    @Column(name = "location_text")
    private String locationText;
    @Column(name = "is_active", nullable = false)
    private boolean active = true;
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        if (id == null) id = UUID.randomUUID();
        if (createdAt == null) createdAt = Instant.now();
    }

    public UUID getId() { return id; }
    public UUID getBusinessId() { return businessId; }
    public void setBusinessId(UUID businessId) { this.businessId = businessId; }
    public UUID getPosterId() { return posterId; }
    public void setPosterId(UUID posterId) { this.posterId = posterId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public String getJobType() { return jobType; }
    public void setJobType(String jobType) { this.jobType = jobType; }
    public String getSalaryRange() { return salaryRange; }
    public void setSalaryRange(String salaryRange) { this.salaryRange = salaryRange; }
    public String getLocationText() { return locationText; }
    public void setLocationText(String locationText) { this.locationText = locationText; }
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
    public Instant getCreatedAt() { return createdAt; }
}
